### ISSUE TITLE
Plugin installation fails with Jenkins 2.x on Ubuntu 14.04

### ISSUE THREAD

**Issue by chris-arnold** [sheehan/job-dsl-gradle-example#64]
I've set up some tests for our DSL repository based on the starting point that this (seriously awesome) project provides. Unfortunately, I'm running into errors with loading plugins that I'm struggling to track down. This could likely be out of the scope of this project, if so - please close this - but I figured I'd start here.

We're running Jenkins version 2.32, Job DSL version 1.53, test harness version 2.17.

On OSX, the plugin installations for the test runs work just fine. Plugins are loaded, the DSL runs, and no exceptions are thrown.

On Ubuntu 14.04 however, the plugins defined in build.gradle do NOT get installed into the test harness. Only the base plugins in the jenkins-for-test/WEB-INF/detached-plugins directory get loaded, many of which are outdated. Our Jenkins server is running on Ubuntu 14.04, so this is problematic for having our tests run prior to the DSL.

For example, the logs show it loads matrix-project 1.4.1, while we define 1.7.1 as a dependency with testPlugin in the build.gradle file. 1.4.1 is included in the detached-plugins. Due to this, many other plugins fail to load due to an outdated dependency.

If, however, I revert to using Jenkins 1.642 like in the source for this project, it DOES work on both OSX and Ubuntu. I fear this may not be ideal though - since we're running 2.32 in reality.

Has anyone seen anything like this before? Is this why this project uses 1.642 despite 2.x being out in the wild for a while now?

The specific output we see when the tests in JobScriptsSpec fail looks like this:
```
   SEVERE: Failed Loading plugin Jenkins Parameterized Trigger plugin v2.32 (parameterized-trigger)
    java.io.IOException: Jenkins Parameterized Trigger plugin v2.32 failed to load.
     - Matrix Project Plugin v1.4.1 is older than required. To fix, install v1.6 or later.
     - Jenkins Subversion Plug-in v1.54 is older than required. To fix, install v2.5.7 or later.
```

If it helps: here's a spec I created to test the plugin loading itself, since we leverage quite a few of them, and I wanted to have a separate means to test this. (Disclaimer: I'm somewhat new to groovy).
```
class PluginLoadSpec extends Specification {

    @Shared
    @ClassRule
    private JenkinsRule jenkinsRule = new JenkinsRule()

    @Shared
    private File outputDir = new File('./build/debug-xml')

    def setupSpec() {
        outputDir.deleteDir()
    }

    void 'test plugins load'() {
        given:
        Jenkins jenkins = jenkinsRule.jenkins
        PluginManager pm = jenkins.pluginManager
        List<PluginWrapper> plugins = pm.plugins
        plugins.each {
            println("Loaded plugin: ${it.shortName}")
            println("Version: ${it.version}")
        }

        when:
        if (!pm.failedPlugins.empty) {
            pm.failedPlugins.each {
                println("PLUGIN LOAD FAILURE: ${it.name}\nCAUSE: ${it.cause}")
            }
        }

        then:
        pm.failedPlugins.empty
    }
}
```

Here's the gradle task I'm using to run the above.
```
task testPluginsLoad(type: Test, overwrite: true) {
    testLogging {
        events 'standard_out'
    }
    include '**/PluginLoadSpec.class'
    exclude '**/JobScriptsSpec.class'
    // Force 'build' directory usage.
    systemProperty 'buildDirectory', project.buildDir.absolutePath
}
```

**Comment by LawYard** [comment]
Hello! 
We have the same error - when `jenkinsVersion` changed from `1.642` to `2.19.1` (or any `2.x`) tests fails with error:

>Caused by:
10:57:43.638 [DEBUG] [TestEventLogger]         javaposse.jobdsl.dsl.DslScriptException: (script, line 72) No signature of method: javaposse.jobdsl.dsl.helpers.triggers.TriggerContext.githubPullRequest() is applicable for argument types: (script$_run_closure2$_closure8$_closure9) values: [script$_run_closure2$_closure8$_closure9@4c560437]

This is actual only for Ubuntu. On my macOS this tests runs successfully.

**Comment by chris-arnold** [comment]
Resolved by PR #70 - thanks! Closing.
